sdk/agent/agent.go

```diff
@@ -167,10 +167,45 @@ type Agent struct {
 	streamerCancel           func()
 }
 
+// Config returns the configuration that the Agent was constructed with.
+func (a *Agent) Config() Config {
+	return Config{
+		ObservationPeriodTime:      a.observationPeriodTime,
+		ObservationPeriodLedgerGap: a.observationPeriodLedgerGap,
+		MaxOpenExpiry:              a.maxOpenExpiry,
+		NetworkPassphrase:          a.networkPassphrase,
+
+		SequenceNumberCollector: a.sequenceNumberCollector,
+		BalanceCollector:        a.balanceCollector,
+		Submitter:               a.submitter,
+		Streamer:                a.streamer,
+		Snapshotter:             a.snapshotter,
+
+		EscrowAccountKey:    a.escrowAccountKey,
+		EscrowAccountSigner: a.escrowAccountSigner,
+
+		LogWriter: a.logWriter,
+
+		Events: a.events,
+	}
+}
+
+// Snapshot returns a snapshot of the agent and its channel.
+func (a *Agent) Snapshot() Snapshot {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	return a.buildSnapshot()
+}
+
 func (a *Agent) snapshot() {
```

Comment:
💡 can we change the unexported snapshot name to something like `takeSnapshot` or `saveSnapsot`? Since the exported (which seems like a getter) has the same name but is doing something different